Modules/DataModel/Sources/PocketCastsDataModel/Public/PlaylistQueryBuilder.swift

```diff
@@ -46,6 +47,9 @@ public class PlaylistQueryBuilder {
         if playlist.manual {
             switch clause {
             case .episode:
+                // Optimization: Filter episodes before deduplication to reduce window function processing
+                let episodeFilter = FeatureFlag.optimizeManualPlaylistQueries.enabled ?
```

Comment:
With the FF on, when showing/hiding the archived episodes, the view detail doesn't update correctly:
- Open a manual playlist with few episodes archived
- Show/hide the archived episodes
- The list updates but the artwork doesn't
- Tap it again and notice the artwork updates based on the previous selection
- Sometimes it doesn't even update the artwork
- Switching off or commenting this line makes it work

https://github.com/user-attachments/assets/35a60d14-b91b-40aa-8cec-37376b0baa29